src/libstd/sys/windows/handle.rs

```diff
@@ -21,7 +21,37 @@ impl Handle {
     pub fn new(handle: HANDLE) -> Handle {
         Handle(handle)
     }
+
+    pub unsafe fn from_fd(fd: c_int) -> Handle {
+        libc::get_osfhandle(fd) as libc::HANDLE
```

Comment:
I think that MSVCRT requires that file descriptors have `close` called on them to free both the backing `HANDLE` as well as MSVCRT resources, so this function may not be able to exist :(